pennylane/vqe/vqe.py

```diff
@@ -108,61 +97,77 @@ def circuit(params):
         tuple: callable functions which evaluate each observable
     """
     if not callable(ansatz):
-        raise ValueError("Could not create quantum circuits. The ansatz is not a callable function.")
+        raise ValueError(
+            "Could not create quantum circuits. The ansatz is not a callable function."
+        )
+
     # TODO: can be more clever/efficient here for observables which are jointly measurable
     circuits = []
-    for obs in observables:
+    for idx, obs in enumerate(observables):
         if not isinstance(obs, Observable):
             raise ValueError("Could not create circuits. Some or all observables are not valid.")
-        @qnode(device=device, interface=interface)
-        def circuit(*params):
-            ansatz(*params)
-            obs.queue()  # HACK: logic for observables assumes that they are in a queue
+
+        def circuit(params, obs=obs):
+            ansatz(*params, wires=range(device.num_wires))
             return expval(obs)
-        circuits.append(circuit)
+
+        qnode = QNode(circuit, device)
+
+        if interface == "tf":
+            qnode = qnode.to_tf()
+        elif interface == "torch":
+            qnode = qnode.to_torch()
+
+        circuits.append(qnode)
+
     return circuits
 
 
-def aggregate(coeffs, qnodes, params, interface="numpy"):
-    """
-    Aggregate a collection of coefficients and expectation values into a single scalar.
+def aggregate(coeffs, qnodes, params):
+    r"""Aggregate a collection of coefficients and expectation values into a single scalar.
 
-    Suppose that the kth :class:`~.QNode` in ``qnodes`` returns the expectation value :math:`\langle O_k \rangle`, and
-    that the kth element of ``coeffs`` is :math:`c_k`.
+    Suppose that the kth :class:`~.QNode` in ``qnodes`` returns the expectation value
+    :math:`\langle O_k \rangle`, and that the kth element of ``coeffs`` is :math:`c_k`.
     Then this function returns the sum :math:`\sum_{k} c_k O_k`.
 
     Args:
         coeffs (Iterable[float]): the coefficients of each summand
-        qnodes (Iterable[:class:`~.QNode`]): callable :class:`~.QNode` functions which return an expectation value
-        params (Iterable[float, tf.Tensor, torch.Tensor]): parameter values to be used as arguments to each
-            individual :class:`~.QNode`
-        interface (str): which interface to use for the :class:`~.QNode`s
+        qnodes (Iterable[:class:`~.QNode`]): Callable :class:`~.QNode` functions which return
```

Comment:
I don't really mind either way - I only changed it because pylint was complaining (it doesn't like that we are using the name of an existing function as an unrelated argument, as it can cause unexpected behaviour). So any naming scheme that avoids redefining existing objects is fine by me.